File: python/insertion-sort.py
def insertionSort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1
        while(j >= 0 and arr[j] > key):
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key
    print(arr)
    return arr

if __name__ == '__main__':
    insertionSort([5, 2, 4, 6, 1, 3])

File: cpp/insertion-sort.cpp
#include <iostream>
#include <vector>

void insertionSort(std::vector<int>& arr) {
    for(int i = 1; i < arr.size(); i++) {
        for(int j = i - 1; j > -1; j--) {
            if(arr[j+1] < arr[j]) {
                std::swap(arr[j+1], arr[j]);
            }
        }
    }
}

int main() {
    std::vector<int> arr = {5, 2, 9, 1, 5, 6};
    insertionSort(arr);
    for(int num : arr) {
        std::cout << num << " ";
    }
    return 0;
}
